skada/_ot.py

```diff
@@ -0,0 +1,213 @@
+# Author: Remi Flamary <remi.flamary@polytechnique.edu>
+#
+# License: BSD 3-Clause
+
+
+import numpy as np
+from sklearn.base import clone
+from sklearn.utils.validation import check_is_fitted
+from .base import DAEstimator
+from .utils import source_target_split
+import ot
+import warnings
+
+
+def solve_jdot_regression(Xs, ys, Xt, base_estimator, alpha=0.5, ws=None, wt=None,
+                          n_iter_max=100, tol=1e-5, verbose=False, **kwargs):
+    """Solve the joint distribution optimal transport regression problem
+
+    Parameters
+    ----------
+    Xs : array-like of shape (n_samples, n_features)
+        Source domain samples.
+
+    ys : array-like of shape (n_samples,)
+        Source domain labels.
+
+    Xt : array-like of shape (m_samples, n_features)
+        Target domain samples.
+    base_estimator : object
+        The base estimator to be used for the regression task. This estimator
+        should solve a least squares regression problem (regularized or not)
+        to correspond to JDOT theoretical regression problem but other
+        approaches can be used with the risk that the fixed point might not converge.
+    alpha : float, default=0.5
+        The trade-off parameter between the feature and label loss in OT metric
+    ws : array-like of shape (n_samples,)
+        Source domain weights (will ne normalized to sum to 1).
+    wt : array-like of shape (m_samples,)
+        Target domain weights (will ne normalized to sum to 1).
+    n_iter_max: int
+        Max number of JDOT alternat optimization iterations.
+    tol: float>0
+        Tolerance for loss variations (OT and mse) stopping iterations.
+    verbose: bool
+        Print loss along iterations if True.as_integer_ratio
```

Comment:
'kwargs' is not documented (it is common to pass additional arguments into a base estimator but still better to mention explicitly)